packages/dd-trace/src/aiguard/client.js

```diff
@@ -0,0 +1,50 @@
+'use strict'
+
+const http = require('http')
+const https = require('https')
+const { URL } = require('url')
+
+function executeRequest (body, opts) {
+  return new Promise((resolve, reject) => {
+    const url = new URL(opts.url)
+    const transport = url.protocol === 'https:' ? https : http
```

Comment:
Have you considered the use of the fetch API? it is much simpler and it is available from node 18? It works to promises, is the same for http and https etc. I think that we are using http in other places because they were implemented to support old node versions.
https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API

Reply:
Migrated to fetch, we didn't want to make a lot of changes but seems to be worth it.

CC: @IlyasShabi that also suggested this change